src/main/java/eu/deltasource/internship/hotel/service/BookingService.java

```diff
@@ -1,21 +1,199 @@
+		this.bookingRepository = bookingRepository;
+		this.roomService = roomService;
+		this.guestService = guestService;
+	}
+
+	/**
+	 * Creates a booking with given booking id, guest id, room id, number of people and dates
+	 *
+	 * @param bookingId      the booking id
+	 * @param guestId        the guest id
+	 * @param roomId         the room id
+	 * @param numberOfPeople the room's capacity
+	 * @param fromDate       the date of accommodation
+	 * @param toDate         the date of leaving
+	 */
+	public void createBooking(int bookingId, int guestId, int roomId, int numberOfPeople, LocalDate fromDate, LocalDate toDate) {
+		validateBookingId(bookingId);
+		if (existsById(bookingId)) {
+			throw new IllegalArgumentException("Booking with id " + bookingId + " already exists.");
+		}
+
+		validateGuestId(guestId);
+		validateRoomId(roomId);
+		Room room = roomService.getRoomById(roomId);
+		if (room.getRoomCapacity() != numberOfPeople) {
+			throw new IllegalArgumentException("Number of people must be equal to room capacity.");
+		}
+
+		validateDates(fromDate, toDate);
+		if (isRoomBookedForPeriod(roomId, fromDate, toDate)) {
+			throw new IllegalArgumentException("Room is already booked for that period.");
+		} else {
+			bookingRepository.save(new Booking(bookingId, guestId, roomId, numberOfPeople, fromDate, toDate));
+		}
+	}
+
+	/**
+	 * Returns all bookings in the repository as unmodifiable list
+	 *
+	 * @return all bookings in the repository as unmodifiable list
+	 */
+	public List<Booking> getAllBookings() {
+		return bookingRepository.findAll();
+	}
+
+	/**
+	 * Returns a booking with particular id
+	 *
+	 * @param id the id of the booking
+	 * @return a booking with particular id
+	 */
+	public Booking getBookingById(int id) {
+		validateBookingId(id);
+		checkIfBookingWithIdDoesNotExists(id);
+
+		return bookingRepository.findById(id);
+	}
+
+	/**
+	 * Updates dates of booking with particular id
+	 *
+	 * @param id       the id of the booking
+	 * @param fromDate the date of accommodation
+	 * @param toDate   the date of leaving
+	 */
+	public void updateBookingDatesById(int id, LocalDate fromDate, LocalDate toDate) {
+		validateBookingId(id);
+		checkIfBookingWithIdDoesNotExists(id);
+		validateDates(fromDate, toDate);
+		Booking booking = getBookingById(id);
+
+		// Check if we are trying to set dates that overlap with current booking dates
+		// if we have booking like 2019.8.1 - 2019.8.5 to be able to set dates like 2019.8.2 - 2019.8.4
+		// if they overlap, the booking dates will be changed otherwise isRoomBookedForPeriod will be called
+		if (!areDatesOverlapping(booking.getFrom(), booking.getTo(), fromDate, toDate)) {
+			if (isRoomBookedForPeriod(id, fromDate, toDate)) {
+				throw new IllegalArgumentException("Room is already booked for that period.");
+			}
+		}
+
+		Booking bookingWithChangedDates = new Booking(id, booking.getGuestId(),
```

Comment:
Let's check if we can use the getters and pass in the `booking` variable to the update